doc/api/stream.md

```diff
@@ -1799,6 +1799,32 @@ Calling `Readable.from(string)` or `Readable.from(buffer)` will not have
 the strings or buffers be iterated to match the other streams semantics
 for performance reasons.
 
+### `stream.addAbortSignal(signal, readable)`
+<!-- YAML
```

Comment:
Member
I really don’t like this. Would much prefer a signal constructor argument

Reply:
The problem is that not every API exposes construction easily or at all and doing this as a constructor argument means all the APIs would need to change

Reply:
Yeah, it's not ideal but I've seen many streams subclasses that do not pass the constructor arguments down consistently. It may still be good to have a constructor argument in addition to this, but I think this api is unfortunately necessary.

Reply:
Maybe we can have both?

Reply:
Happy to add both

Reply:
I'll keep this one `addAbortSignal` and follow up with another PR with the constructor argument if that's OK.

Reply:
If we keep this I think we need a way to remove the signal.

Reply:
continued in discussion below.